modules/Bio/EnsEMBL/Compara/RunnableDB/Flatfiles/MySQLImportHomologies.pm

```diff
@@ -0,0 +1,164 @@
+    my %attribs;
+    my $primary_key = 'homology_id';
+    foreach my $f ( @$attrib_files ) {
+        open( my $fh, '<', $f ) or die "Cannot open $f for reading";
+        my $header = <$fh>;
+        my @header_cols = split( /\s+/, $header );
+        die "No $primary_key found in $f - please check file header line\n" unless grep {$_ eq $primary_key} @header_cols;
+        while ( my $line = <$fh> ) {
+            my $row = map_row_to_header($line, \@header_cols);
+            my $primary_id = $row->{$primary_key};
+            delete $row->{$primary_key};
+            foreach my $attrib_name ( keys %$row ) {
+                $attribs{$primary_id}->{$attrib_name} = $row->{$attrib_name};
+            }
+        }
+        close $fh;
+    }
+
+    $self->param('attribs', \%attribs);
+}
+
+sub write_output {
+    my $self = shift;
+
+    my $filename = $self->param_required('homology_flatfile');
+    my $attribs  = $self->param('attribs');
+
+    # open homology file
+    open(my $hom_fh, '<', $filename) or die "Cannot open $filename for reading";
+    my $header_line = <$hom_fh>;
+    my @header_cols = split(/\s+/, $header_line);
+
+    # open output files - one per table in tmp space
+    my $homology_csv = $self->worker_temp_directory . '/homology.txt';
+    open(my $h_csv, '>', $homology_csv) or die "Cannot open $homology_csv for writing";
+    my $homology_member_csv = $self->worker_temp_directory . '/homology_member.txt';
+    open(my $hm_csv, '>', $homology_member_csv) or die "Cannot open $homology_member_csv for writing";
+
+    # iterate over homology input and format it for later mysqlimport
+    my $h_count;
+    my $homology_id_start = $self->param_required('homology_id_start');
+    while ( my $line = <$hom_fh> ) {
+        my $row = map_row_to_header($line, \@header_cols);
+
+        # add the attribs
+        foreach my $attrib_name ( 'goc_score', 'wga_coverage', 'is_high_confidence' ) {
+            my $this_attrib = $attribs->{$row->{homology_id}}->{$attrib_name};
+            $row->{$attrib_name} = defined $this_attrib ? $this_attrib : '\N';
+        }
+
+        my $this_homology_id = $homology_id_start + $h_count;
+        my ( $homology_row, $homology_member_rows ) = $self->split_row_for_homology_tables($row, $this_homology_id);
+        print $h_csv $homology_row;
+        print $hm_csv $homology_member_rows;
+        $h_count++;
+    }
+    close $h_csv;
+    close $hm_csv;
+    print "Files written to: $homology_csv & $homology_member_csv\n" if $self->debug;
+
+    # mysqlimport the data
+    my $this_dbc = $self->compara_dba->dbc;
+    my $user = $this_dbc->username;
+    my $pass = $this_dbc->password;
+    my $host = $this_dbc->host;
+    my $port = $this_dbc->port;
+    my $dbname = $this_dbc->dbname;
+    my $import_cmd = join(' ', 
+        "mysqlimport --host=$host --port=$port --user=$user --password=$pass",
+        "--local --lock-tables=0 --ignore --fields-terminated-by=','",
+        $dbname
+    );
```

Comment:
I didn't test the `bulk_insert` method as it stands. I think the ideal scenario, if all the data is already loaded in memory, would be to "pipe" it to a `LOAD DATA` (or `mysqlimport`) command, since dumping it into files and then reading them was about a 5% slower in my tests with `mysqldump ... > file.txt` + `mysqlimport ... file.txt`.

Reply:
The pipe into `LOAD DATA` is a very good idea. Can I create a Jira ticket or do you want to do it ?

Reply:
Well, actually, this trick could be used here ! Instead of writing a file and running `mysqlimport`, we could pipe into `LOAD DATA`

Reply:
Can this be an extra ticket later? I think there's enough happening in this PR

Reply:
Also, I'm not sure if it would handle the 2 different tables at the same time

Reply:
Yes. the ticket is meant for a later development, not this PR :)